src/layer/__tests__/cache.ts

```diff
@@ -0,0 +1,115 @@
+import 'jest-setup';
+import axios from 'axios';
+import MockAdapter from 'axios-mock-adapter';
+import makeServiceWorkerEnv from 'service-worker-mock';
+import fetch from 'node-fetch';
+
+import { constructFixtureFindTiles } from './fixtures.findTiles';
+import { constructFixtureGetMap } from './fixtures.getMap';
+import { ApiType } from 'src';
+import { setAuthToken } from 'src/auth';
+
+const mockNetwork = new MockAdapter(axios);
+
+const EXAMPLE_TOKEN = 'TOKEN111';
+
+describe('Testing caching', () => {
+  beforeEach(() => {
+    Object.assign(global, makeServiceWorkerEnv(), fetch);
+    jest.resetModules();
+  });
+
+  it('should fetch a request and cache it, where 2nd request is served from the cache', async () => {
+    const {
+      fromTime,
+      toTime,
+      bbox,
+      layer,
+      mockedResponse,
+      expectedResultTiles,
+      expectedResultHasMore,
+    } = constructFixtureFindTiles({});
+    const requestsConfig = {
+      cache: {
+        expiresIn: 60,
+      },
+    };
+    mockNetwork.reset();
+    mockNetwork.onPost().replyOnce(200, mockedResponse);
+    mockNetwork.onPost().replyOnce(200, mockedResponse);
+
+    const responseFromMockNetwork = await layer.findTiles(bbox, fromTime, toTime, null, null, requestsConfig);
+    const fromCacheResponse = await layer.findTiles(bbox, fromTime, toTime, null, null, requestsConfig);
+
+    expect(mockNetwork.history.post.length).toBe(1);
+    expect(fromCacheResponse.tiles).toStrictEqual(expectedResultTiles);
+    expect(fromCacheResponse.hasMore).toBe(expectedResultHasMore);
+    expect(fromCacheResponse).toStrictEqual(responseFromMockNetwork);
+  });
+
+  it('should make a 2nd request after the cache has expired', async () => {
+    const { fromTime, toTime, bbox, layer, mockedResponse } = constructFixtureFindTiles({});
+    const requestsConfig = {
+      cache: {
+        expiresIn: 1,
+      },
+    };
+    mockNetwork.reset();
+    mockNetwork.onPost().replyOnce(200, mockedResponse);
+    mockNetwork.onPost().replyOnce(200, mockedResponse);
+
+    await layer.findTiles(bbox, fromTime, toTime, null, null, requestsConfig);
+    await new Promise(r => setTimeout(r, 1100));
+    await layer.findTiles(bbox, fromTime, toTime, null, null, requestsConfig);
+
+    expect(mockNetwork.history.post.length).toBe(2);
+  });
+
+  it('test that no responses are cached', async () => {
+    const { fromTime, toTime, bbox, layer, mockedResponse } = constructFixtureFindTiles({});
+    mockNetwork.reset();
+    mockNetwork.onPost().replyOnce(200, mockedResponse);
+    mockNetwork.onPost().replyOnce(200, mockedResponse);
+
+    await layer.findTiles(bbox, fromTime, toTime);
+    await layer.findTiles(bbox, fromTime, toTime);
+
+    expect(mockNetwork.history.post.length).toBe(2);
```

Comment:
Since caching is set to half an hour by default, I would expect this test to fail. What am I missing?

Reply:
We never cached `findTiles`, and since `cache` is not set in this test there should be no cache

Reply:
As discussed ,  `cacheConfig` is passed to `getAxiosReqParams` in order to make it more explicit what is cached by default and what is not.

Reply:
Nice!